### DISCUSSION TITLE
Target PCB Design

### DISCUSSION THREAD

**Discussion by olsson82** [olsson82/olshot#4] [Announcements]
The PCB Design for the Shooting targets is in design mode.

## Arduino Chip
They will use ATMega328-AU Chip, same as in Arduino Nano. So you need to know some SMB Soldering.
You will be able to program them with an ISP programmer, the PCB card will be designed for it.

## Built in charger
There will be an built in charger for LiPo/Lithium batterys.

## NRF Module
The Wireless module will not be built in, there will be an socket so you can insert it. This is so you can choose to have both type. (The one with internal antenna, and the one without.)

## 4 Leds
There will be 4 leds on the board. One is for the target signal, and one is for power/low battery warning.
The other two is for the charger. One is for when you plug in the charger, and the other is to know when it's charge.

**Comment by olsson82** [comment]
The first design is done for the target sensor.
![shootingsensor_back](https://user-images.githubusercontent.com/122672087/212536472-4af14b09-78e3-412f-953b-31fb34bdf01e.png)
![shootingsensor_front](https://user-images.githubusercontent.com/122672087/212536474-7c81ba49-6e64-49bf-b0b1-639cad596770.png)

Changes can be made during the work of the system.

**Comment by olsson82** [comment]
This is the final design (I think) of the PCB. Going to check everything again so it correct. Also the JCLPCB components so the footprint is correct.

![targetsensor-info](https://user-images.githubusercontent.com/122672087/213912753-92d90810-8bd9-4be7-8392-09c033855405.png)

The shake switch is connected with jcl connector. For me i don't want the pcb with battery on the target. So with a wire i place the sensor on the target and connect it on the board. The external sensor is the same as the shake sensor but you can also have the target light on there.

**Comment by h8dis** [comment]
Hi cool project, I got the pcbs printed and assembled. I'm having trouble burning the bootloader, are you sure all connections are right or am I doing something  wrong

**Reply by olsson82** [reply]
Hi what type of programmer are you using for burn the bootloader ?
Also please know that the main unit is still in work. Right now the sensors are only one that is done.

**Reply by h8dis** [reply]
I worked it out what i was doing wrong and have the sensors done now.  How's the main system coming along
 I love your project I'm so keen to get it finished and working . Thanks

**Reply by olsson82** [reply]
Glad it worked out. Im working on redesign, and going to make sure everything will work. I hope i can send it to manufacture next week.

**Reply by h8dis** [reply]
Awesome keep up the good work, I'm excited to check it out

**Reply by olsson82** [reply]
Thanks :) The manufacture is almost done, so i will have it next week. Then there will be some testing before i release the files for the main unit.

**Reply by olsson82** [reply]
![IMG_3313](https://github.com/olsson82/olshot/assets/122672087/0d0c8b0e-81c3-4b8c-a7c9-7457669bb6b0)

Here is the main unit. I will upload the kicad files later today or tomorrow.

**Comment by h8dis** [comment]
That's awesome well done.. I'm gonna get some made this week. Keep up the
good work. I'm so excited I can't wait to get the whole project working ..

**Reply by olsson82** [reply]
Check on the wiki here on github so you know how the j6 j7 and j8 is used when uploading code. They need to be unconnected during upload and burning bootloader, due to microsd card function.

**Comment by h8dis** [comment]
What nextion screen do you recommend to work with the project so that I cam
get a head start and get one ordered?

**Reply by olsson82** [reply]
I use the basic 7" screen. No need to any advanced screens.

**Comment by h8dis** [comment]
Hi mate I ordered the main unit pcb on jlcpcb using the files you provided.
They sent me this email, I'm not sure what to do.. do you understand what
this means and what should I do?

**Reply by olsson82** [reply]
I think you forgot the mail from them.

**Comment by h8dis** [comment]
Hi sorry I attached a screenshot bot sure if you got it
.



The email says

Hi Sir or madam,



Well got your order with many thanks.



Sorry to bother you, but there is one thing that we want to confirm with
you about your SMT order before proceeding.



As shown below, seems the packages of the parts ( J9) cannot match the
corresponding pads on the PCB board, so we are unable to assemble them.

For this case there are 3 options for your choose as follows:

A- we enable the replace part button for you that you can re-select the
corrected part


B- we leave this parts unpopulated and refund cost of the mismatch parts
back to you to proceed and can’t ship the unsoldered parts with PCB due to
the customs clearance issue,

C- If you need to change gerber file or BOM or CPL file, we can help you
cancel it and customer can place new order (there is no replace gerber file
button for SMT order)

Pls consider and advise which case suits you, A or B or C??



Your early reply will be highly appreciated, thank you so much!

**Reply by olsson82** [reply]
Looks like the J9 component has been changed. As it not match the footprint.

Reply to them with answer A, they will enable a button for you to change component.

on there you can search for a new matching component to use.

**Comment by h8dis** [comment]
Thanks mate I got it sorted. My main unit pcbs should arrive within the
week I think, how is the code coming along?

**Reply by olsson82** [reply]
The code is going good, still have some things to work out, but it's getting closer to be done. Stay tune!

**Comment by h8dis** [comment]
Hi not meaning to rush your brilliant work  i  I have received my pcbs and
pretty keen to put them into action. Is there  a version of code I can put
on them, I don't mind if has bugs can update later. Just thought I'd ask
coz I'm excited to try them. Either way thank you

**Reply by olsson82** [reply]
I understand you.
Im going to upload "BETA" version of the code.
Everything but the clock function is there. The clock function is to save the date and time to highscore, but i need to work on that little more. (Some memory issues)

Also it's not so nice design yet, but it's a work in progress :)

Im going to write all info that is needed in the wiki about the main unit code. Then the files will be online :)

**Reply by olsson82** [reply]
The code and the nextion design is now uploaded, also read the wiki about the main unit code.

**Comment by h8dis** [comment]
Awesome, thank you so much!  U are the best..   what crystal is needed for
y2?

**Reply by olsson82** [reply]
Thanks :)

Oh sorry, missed that in the bom list. It's 32.768kHz crystal.

**Comment by DaannD** [comment]
Hello! What kind of LiPo do you use? What voltage? What kind of shake sensor do you use? What nextion display do you use?

**Reply by olsson82** [reply]
Hi sorry about the late answer.

For the lipo i use 3.7V 2000 mAh
For the sensor i use analog piezo sensor (need to solder a resistor for it to be use, you can read about it in the wiki)
And for the Nextion display i use the Basic 7" display.
